pkg/collector/corechecks/sbom/processor.go

```diff
@@ -295,6 +296,13 @@ func (p *processor) processImageSBOM(img *workloadmeta.ContainerImageMetadata) {
 			}
 		}
 
+		repoDigests := make([]string, 0, len(img.RepoDigests))
+		for _, repoDigest := range img.RepoDigests {
+			if strings.HasPrefix(repoDigest, repo+"@sha256:") {
```

Comment:
I feel like if we remove the repo part it might be a bit confusing. It means that we can have a payload such as:
```
sbom{
  id: gcr.datadoghq.com/agent@sha256:imageconfighash,
  repodigests: [
    hash1, // repodigest hash on amazon ecr,
    hash2, // repodigest hash on gcr,
  ]
}
```
What do you think ?

Reply:
It’s not possible to have hashes for different repo as in your example because of the the `if strings.HasPrefix(repoDigest, repo+"@sha256:") {` that filters the repo digests to keep only the ones related to the currently considered repo.

I stripped the repo part from the digests to be symmetric with what we already do for the repo tags.

But I agree that we can keep the full repo digests if it ease anything.

Reply:
Fixed.
I now keep the full repo digest.
Here is an example of what we get:

```shell
fakeintakectl --url http://… filter sbom --id ghcr.io/datadog/apps-nginx-server@sha256:de5c472beb90fb7b4726b1a56b15076b06f2857f8836ed4535f09ead70b37582
```
```json
[
  {
    "type": 1,
    "id": "ghcr.io/datadog/apps-nginx-server@sha256:de5c472beb90fb7b4726b1a56b15076b06f2857f8836ed4535f09ead70b37582",
    "repo_tags": [
      "main"
    ],
    "inUse": true,
    "dd_tags": […],
    "Sbom": {…},
    "status": 1,
    "repo_digests": [
      "ghcr.io/datadog/apps-nginx-server@sha256:625f62fb6ebb306741285fbb600b62d9fa93a11215b9f3c9a51393b04913de2c"
    ]
  },
```